twenty-first/src/util_types/storage_vec.rs

```diff
@@ -10,13 +10,54 @@ use serde::{de::DeserializeOwned, Serialize};
 pub type Index = u64;
 
 pub trait StorageVec<T> {
+    /// check if collection is empty
     fn is_empty(&self) -> bool;
+
+    /// get collection length
     fn len(&self) -> Index;
+
+    /// get single element at index
     fn get(&self, index: Index) -> T;
+
+    /// get multiple elements matching indices
     fn get_many(&self, indices: &[Index]) -> Vec<T>;
+
+    /// get all elements
     fn get_all(&self) -> Vec<T>;
+
+    /// set a single element.
     fn set(&mut self, index: Index, value: T);
+
+    /// set multiple elements.
+    fn set_many(&mut self, key_vals: impl IntoIterator<Item = (Index, T)>);
+
+    /// set all elements.
+    fn set_all(
+        &mut self,
+        vals: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = (Index, T)>>,
+    );
```

Comment:
There is `set_all_by_array()` for that. 

However, looking at it with fresh eyes, I see that current impl of `set_all` does not enforce that Index is unique and therefore it is not really set_all, but rather kind of a dup of set_many.  And enforcing uniqueness is potentially expensive for large inputs.

So now I am thinking to remove `set_all` as it currently exists, and rename set_all_by_array() to `set_all`, and it will just call `set_many` with array indexes converted to `Index`.